src/GraphQL/Validation/DocumentValidator.cs

```diff
@@ -118,12 +118,36 @@ private async Task<(IValidationResult validationResult, Variables variables)> Va
                     await new BasicVisitor(visitors).VisitAsync(context.Document, new BasicVisitor.State(context)).ConfigureAwait(false);
                 }
 
+                if (context.HasErrors)
+                {
+                    return new ValidationResult(context.Errors);
```

Comment:
Do you want fast return here even without reading variables? After this return the control will flow into `BuildExecutionContext` and `foreach (var listener in options.Listeners)` loop. That code can use variables in one or another way.

Reply:
I expected such an answer. Then documentation about variables should clear say that variables are not parsed because ... (your text from the comment above).